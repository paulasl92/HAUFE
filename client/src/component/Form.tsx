import  { useState } from "react";
import { Link, useHistory } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";

import { signIn , signUp} from "../redux/actions/user-action";

const Form = ( {formInformation : formInformation}) => {
    const dispatch = useDispatch();
    const history = useHistory();
    const { error } = useSelector((state) => state.authentication);
  
    const [email, setEmail] = useState("");
    const [password, setPassword] = useState("");
  
    const handleEmailChange = (event) => {
      setEmail(event.target.value);
    };
    const handlePasswordChange = (event) => {
      setPassword(event.target.value);
    };

  const handleSignUpClick = (event) => {
    event.preventDefault();
    dispatch(signUp({ email, password }, history));
  };

  const handleSignInClick = (event) => {
    event.preventDefault();
    dispatch(signIn({ email, password }, history));
  };

  return (
    <div className="">
      <div className="">
        <input
          type="text"
          placeholder="email"
          value={email}
          onChange={handleEmailChange}
        />
        <input
          type="password"
          placeholder="password"
          value={password}
          onChange={handlePasswordChange}
        />
        {error !== null ? <div className="error">{error}</div> : null}
        <button
          className=""
          onClick={ formInformation.buttonEvent == "UP" ? handleSignUpClick : handleSignInClick }
        >
          {formInformation.buttonText}
        </button>
        <Link to={formInformation.link}>{formInformation.linkText}</Link>
      </div>
    </div>
  );
};

export default Form;