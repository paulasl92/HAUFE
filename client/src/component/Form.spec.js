import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import  Form  from './Form';
import Adapter from 'enzyme-adapter-react-16';
configure({adapter: new Adapter()});

const props = {
    formInformation : {
        buttonText: "Login",
        buttonEvent: "IN",
        link: "/signup",
        linkText: "Don't have an account?",
        buttonRedirect: "Register!",
        handleClick: async (email, password) => {
          const res = await dispatch(signIn({ email, password }, history));
          setError(res);
        }
      },
    form: {
      email: 'Email',
      password: 'Password',
      emailRequired: 'Email is required',
      emailNotValid: 'Email is not valid',
      passwordRequired: 'Password is required',

    },
    Form: () => {},
   };
 
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
describe('>>>Form - Shallow Render REACT COMPONENTS', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = itmes => shallow(<Form {...itmes} />);
    });

// Create 'snapshot'
 it('+++ render Form correctly, snapshot', () => {
  const component = wrapper(props);
  expect(toJson(component)).toMatchSnapshot();
 });

// Check render comonent without errors
 it('+++ render the DUMB component', () => {
  expect(wrapper(props).length).toEqual(1);
 });

// Simple behavior for fill in the fields
 it('+++ simulate change email', () => {
  const component = wrapper(props);
  const eventEmail = {
   target: { name: 'email', value: 'test@test.com' }
  };
  component.find('[name="email"]').simulate('change', eventEmail);

  expect(component.find('[name="email"]').prop('value')).toEqual('test@test.com');

 });
 
});
