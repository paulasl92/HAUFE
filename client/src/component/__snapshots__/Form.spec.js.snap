// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Form - Shallow Render REACT COMPONENTS +++ render Form correctly, snapshot 1`] = `
<div>
  <div
    className="errorContainer"
  >
    <div
      className="error"
    >
      Please enter the email
    </div>
    <div
      className="error"
    >
      Password needs to be at least six characters
    </div>
  </div>
  <div
    className="form"
  >
    <div
      className="login login__form"
    >
      <div
        className="container cont__login"
      >
        <input
          className="email input"
          name="email"
          onChange={[Function]}
          placeholder="email"
          required={true}
          type="text"
          value=""
        />
        <input
          className="password input"
          onChange={[Function]}
          placeholder="password"
          required={true}
          type="password"
          value=""
        />
        <button
          className="btn btn__login"
          disabled={true}
          onClick={[Function]}
        >
          Login
        </button>
      </div>
    </div>
    <div
      className="cta"
    >
      <h2
        className="cta__heading"
      >
        Don't have an account?
      </h2>
      <div>
        <Link
          to="/signup"
        >
          <span
            className="btn__general btn cta__btn cta__btn--text"
          >
            Register!
          </span>
        </Link>
      </div>
    </div>
  </div>
</div>
`;
